File: scripts/download_images.py
import os
import gdown

def download_folder(folder_url, output_dir):
    # Create output directory if it doesn't exist
    os.makedirs(output_dir, exist_ok=True)
    
    # Download the folder
    gdown.download_folder(
        url=folder_url,
        output=output_dir,
        quiet=False,
        use_cookies=False
    )

if __name__ == "__main__":
    # Your Google Drive folder URL
    FOLDER_URL = "https://drive.google.com/drive/folders/1y_ATn27CofUMOfPyVo7WqPP5EogXOkUL"
    OUTPUT_DIR = "public/images/products"
    
    print(f"Downloading images from {FOLDER_URL} to {OUTPUT_DIR}...")
    download_folder(FOLDER_URL, OUTPUT_DIR)
    print("Download completed!")

File: scripts/download-images.js
const fs = require('fs');
const path = require('path');
const { google } = require('googleapis');

// Your Google Drive folder ID
const FOLDER_ID = '1y_ATn27CofUMOfPyVo7WqPP5EogXOkUL';
const OUTPUT_DIR = path.join(__dirname, '../public/images/products');

// Create output directory if it doesn't exist
if (!fs.existsSync(OUTPUT_DIR)) {
  fs.mkdirSync(OUTPUT_DIR, { recursive: true });
}

async function downloadImages() {
  try {
    // Create a new instance of the drive client
    const drive = google.drive({ version: 'v3', auth: process.env.GOOGLE_API_KEY });

    // List all files in the folder
    const res = await drive.files.list({
      q: `'${FOLDER_ID}' in parents`,
      fields: 'files(id, name, mimeType, webContentLink, webViewLink)'
    });

    const files = res.data.files;
    
    if (files.length === 0) {
      console.log('No files found in the specified folder.');
      return;
    }

    console.log('Downloading files...');
    
    // Download each file
    for (const file of files) {
      if (file.mimeType.startsWith('image/')) {
        const dest = path.join(OUTPUT_DIR, file.name);
        const destStream = fs.createWriteStream(dest);
        
        console.log(`Downloading ${file.name}...`);
        
        const res = await drive.files.get(
          { fileId: file.id, alt: 'media' },
          { responseType: 'stream' }
        );
        
        await new Promise((resolve, reject) => {
          res.data
            .on('end', () => {
              console.log(`✅ Downloaded ${file.name}`);
              resolve();
            })
            .on('error', (err) => {
              console.error(`❌ Error downloading ${file.name}:`, err);
              reject(err);
            })
            .pipe(destStream);
        });
      }
    }
    
    console.log('All files downloaded successfully!');
  } catch (error) {
    console.error('Error:', error.message);
  }
}

// Run the function
downloadImages();
